src/EnergyPlus/GroundTemperatureModeling/KusudaAchenbachGroundTemperatureModel.cc

```diff
@@ -90,7 +90,7 @@ KusudaGroundTempsModel::KusudaGTMFactory(EnergyPlusData &state, GroundTempObjTyp
     // New shared pointer for this model object
     std::shared_ptr<KusudaGroundTempsModel> thisModel(new KusudaGroundTempsModel());
```

Comment:
FWIW I know you said this pointer stuff is "too cute by half". It's not. Pointers are the way to leverage inheritance and polymorphism, which I think should be possible to do here, since these are different implementations of ground temperature models that should all have similar APIs. The only thing that may be too cute by half here is the use of `shared_ptr`, which I am not sure is the right wrapper (I think `unique_ptr` might be more appropriate) or even necessary (memory management of these objects is not so complicated that it should require these wrappers).